### ISSUE TITLE
Allow profile dropdown to be closed when clicking anywhere outside of the dropdown card. 

### ISSUE THREAD

**Issue by MingCWang** [MingCWang/deiseval#93]
**Describe the bug**
The current dropdown after clicking the user profile button only closes if the user clicks on the profile button again, for a more streamlined user experience, we want to allow the users to close the dropdown when clicked anywhere else other than the dropdown. 

**To Reproduce**
1. Sign in
2. Click on the profile button on the top right of the navigation bar. 

**Additional context**
You can find the dropdown in the `frontend/src/components/NavBar`

**Comment by RahalBhupathi** [comment]
I am having some difficulty testing, since I can't manually trigger the profile downdown without signing in first. However, I added an event listener in NavBar.jsx that should fix the issue.

**Comment by MingCWang** [comment]
@RahalBhupathi Thank you for your contribution! Have you tried changing the state to `true` on line 17?  This should allow the dropdown to appear on page load, you should be able to test the closing functionality. 

```javascript
const [profileClicked, setProfileClicked] = useState(false);
```

Another way would be to create your own OAuth2.0 credentials.

**Comment by RahalBhupathi** [comment]
Thanks for the help, I moved my changes to ProfileDropdown.jsx and it seems to be working now. The profile menu closes whenever I click anywhere outside, but the dropdown itself seems to have shifted for some reason. I made some small changes to the .container in the ProfileDropdown.css file to account for this. 
![chrome_DhNwIKfprz](https://github.com/MingCWang/deiseval/assets/113255492/927f8502-94b5-4511-8cf0-ac6e4d551255)
